lib/Index/IndexRecord.cpp

```diff
@@ -421,14 +423,15 @@ static void addModuleDependencies(ArrayRef<ImportedModule> imports,
           bool withoutUnitName = true;
           if (FU->getKind() == FileUnitKind::ClangModule) {
             auto clangModUnit = cast<ClangModuleUnit>(LFU);
-            if (!clangModUnit->isSystemModule() || indexSystemModules) {
-              withoutUnitName = false;
+            if ((!clangModUnit->isSystemModule() || indexSystemModules)) {
+              withoutUnitName = !indexClangModules;
```

Comment:
I'm not sure what effect, if any, not including the unit name has? Any ideas?

Reply:
The main thing is modules that are not reachable from the "main units" will be dropped - for our case this should be ok since we'll use the explicit outputs list instead.